MIGRATION.md

```diff
@@ -0,0 +1,91 @@
+# Migrating to the `cloudant-java-sdk` library
+This document is to assist in migrating from the `java-cloudant` (package: `com.cloudant.cloudant-client`) to the newly supported [`cloudant-java-sdk`](https://github.com/IBM/cloudant-java-sdk) (package: `com.ibm.cloud.cloudant`).
+
+## Initializing the client connection
+There are several ways to create a client connection in `cloudant-java-sdk`:
+1. [Environment variables](https://github.com/IBM/cloudant-java-sdk#authentication-with-environment-variables)
+2. [External configuration file](https://github.com/IBM/cloudant-java-sdk#authentication-with-external-configuration)
+3. [Programmatically](https://github.com/IBM/cloudant-java-sdk#programmatic-authentication)
+
+[See the README](https://github.com/IBM/cloudant-java-sdk#code-examples) for code examples on using environment variables.
+
+## Other differences
+1. Fetching the Database object first before performing additional operations is not required. For example, in the case of updating a document you would first call `getDocument` to fetch and then `putDocument` to update.
+1. Model classes are used instead of POJOs.
+1. Sending and receiving byte responses is available for some of our operations.  See [the Raw IO section](https://github.com/IBM/cloudant-java-sdk#raw-io) of `cloudant-java-sdk` README for more details.
+1. There is no pagination support for views.
```

Comment:
I think it would be worth pointing to the pagination recipes - I think the best way to do this would be to add the `Pagination` front-matter to the API reference and then link to there as it is useful outside of the context of migration in any case. I'll raise another item for that work.

For now maybe a statement of "examples coming soon" or something?

Reply:
Done in https://github.com/cloudant/java-cloudant/pull/518/commits/4a283b0b7a09350f8d6798a5ec3b08e9d0358777